### DISCUSSION TITLE
上升/空格 与 下降/shift 不符合基岩版机制

### DISCUSSION THREAD

**Discussion by WhiteElephant-abc** [TouchController/TouchController#114] [输入控制逻辑]
在国际基岩版1.21.51.02中，上升与空格是同一个按键，下降与shift是同一个按键，如果是TC里的这种情况，即使将上升与空格重叠，也会导致例如`双击空格后，需要抬起手指再按下才能向上飞行`的情况
且国际基岩版1.21.51.02中，飞行按键与游泳按键贴图都为飞行

**Comment by fifth-light** [comment]
其实很多时候我还是偷了懒，直接把触控输入映射到了 Minecraft 原版处理输入事件的地方。这样好处就是简单，例如双击 w 疾跑，双击空格飞行等都相当于是原版送的。如果真要一比一复刻基岩版逻辑，那么得更深一步调用实体的方法了。
